march_hardware/src/joint.cpp

```diff
@@ -18,35 +16,24 @@ Joint::Joint(std::string name, int net_number) : name_(std::move(name)), net_num
 {
 }
 
-Joint::Joint(std::string name, int net_number, bool allow_actuation, std::unique_ptr<IMotionCube> imc)
-  : name_(std::move(name)), net_number_(net_number), allow_actuation_(allow_actuation), imc_(std::move(imc))
+Joint::Joint(std::string name, int net_number, bool allow_actuation, std::shared_ptr<MotorController> controller)
+  : name_(std::move(name))
+  , net_number_(net_number)
+  , allow_actuation_(allow_actuation)
+  , controller_(std::move(controller))
 {
 }
 
-Joint::Joint(std::string name, int net_number, bool allow_actuation, std::unique_ptr<IMotionCube> imc,
-             std::unique_ptr<TemperatureGES> temperature_ges)
+Joint::Joint(std::string name, int net_number, bool allow_actuation, std::shared_ptr<MotorController> controller,
+             std::shared_ptr<TemperatureGES> temperature_ges)
```

Comment:
Why has this been changed to a `shared_ptr`?

Reply:
See comments on the others